### PR TITLE
Port :erlang.setelement/3 to JS

### LINKED ISSUE #359: Port :erlang.setelement/3 to JS

### PR DESCRIPTION
**Pull request by Sorc96**
Closes #359

Currently checks the length of the tuple's data directly to determine whether the index is in or out of range. Could be changed to use `:erlang.tuple_size/1` after #331 is merged.

### CODE REVIEW

**Comment by bartblast** [comment]
I've just added "Quality Checks Before Submitting" section here: https://hologram.page/reference/contributing, though to get the updated checks you'll need to sync with the `dev` branch.

File: assets/js/erlang/erlang.mjs [hunk 1]
```diff
@@ -881,6 +881,35 @@ const Erlang = {
...
+      );
+    }
+
+    if (!Type.isTuple(tuple)) {
+      Interpreter.raiseArgumentError(
+        Interpreter.buildArgumentErrorMsg(2, "not a tuple"),
+      );
+    }
+
+    if (index.value <= 0 || index.value > tuple.data.length) {
```
Comment (bartblast) [inline assets/js/erlang/erlang.mjs]: `index.value <= 0n`
will be marginally faster

File: test/elixir/hologram/ex_js_consistency/erlang/erlang_test.exs [hunk 2]
```diff
@@ -2010,6 +2010,36 @@ defmodule Hologram.ExJsConsistency.Erlang.ErlangTest do
     end
   end
 
+  describe "setelement/3" do
+    test "replaces the element at the given index" do
```
Comment (bartblast) [inline test/elixir/hologram/ex_js_consistency/erlang/erlang_test.exs:2014]: Could you test boundary conditions?
* first item
* middle item
* last item

File: test/javascript/erlang/erlang_test.mjs [hunk 3]
```diff
@@ -2767,6 +2767,47 @@ describe("Erlang", () => {
     });
   });
 
+  describe("setelement/3", () => {
+    const setelement = Erlang["setelement/3"];
+
+    it("replaces the element at the given index", () => {
```
Comment (bartblast) [inline test/javascript/erlang/erlang_test.mjs:2773]: Could you test boundary conditions?
* first item
* middle item
* last item

**Comment by bartblast** [comment]
> Currently checks the length of the tuple's data directly to determine whether the index is in or out of range. Could be changed to use `:erlang.tuple_size/1` after #331 is merged.

I'm ok to do this directly in this function as a performance optimization (one less input validation - since the other function validates its input as well).

File: assets/js/erlang/erlang.mjs (hunk 1 above)
Comment (Sorc96) [reply assets/js/erlang/erlang.mjs]: Thanks, that was an oversight on my part.

**Comment by bartblast** [comment]
Thanks for the updates! :)
